File: typescript/series/series.ts
class Series {
  public readonly digits: Array<number>;

  constructor(series: string) {
    this.digits = series
      .split("")
      .map((digit: string) => Number.parseInt(digit));
  }

  slices(width: number): Array<Array<number>> {
    const numberOfDigits = this.digits.length;
    if (width > numberOfDigits) {
      throw new Error();
    }

    const numberOfSlices = numberOfDigits - width + 1;
    const sliceStarts = [...Array(numberOfSlices).keys()];

    return sliceStarts.map((sliceStart: number) =>
      this.digits.slice(sliceStart, sliceStart + width)
    );
  }
}

export default Series;

File: clojure/series/src/series.clj
(ns series)

(defn- slice [slices source length]
  (if (> length (count source))
    slices
    (slice
     (conj slices (subs source 0 length))
     (subs source 1)
     length)))

(defn slices
  [source length]
  (if (= 0 length)
    '[""]
    (slice '[] source length)))
